lib/travis/build/script/shared/directory_cache/s3.rb

```diff
@@ -94,11 +93,11 @@ def push
           end
 
           def fetch_url(branch = group)
-            url('GET', prefixed(branch), expires: fetch_timeout)
+            url('GET', prefixed(branch, '.tbz'), expires: fetch_timeout)
```

Comment:
See https://github.com/travis-ci/casher/pull/12#discussion_r33323761 for a little more context. I think you're right that we need more thought here; it would be great to test it on staging.